scripts/testing/programs/match_template/test_template_reproducibility.py

```diff
@@ -0,0 +1,265 @@
+                if not exists(mip_file):
+                    raise FileNotFoundError(f"MIP file not found at {mip_file}")
+
+                # Get the histogram file path and extract the threshold value
+                hist_file = join(config['output_file_prefix'], 'hist.txt')
+                threshold_value = extract_threshold_value(hist_file)
+
+                mip_filenames.append(mip_file)
+                hist_filenames.append(hist_file)
+                threshold_values.append(threshold_value)
+
+                # Print threshold value only for the first replicate
+                if replicate == 0:
+                    print(f"Threshold value: {threshold_value:.6e}")
+
+                print(f"Completed replicate {replicate+1}, time: {elapsed_time[replicate]:.2f}s")
+
+            except Exception as e:
+                print(f"Error during replicate {replicate+1}: {str(e)}")
+                continue
+
+        # Check if we have all three replicates
+        if len(mip_filenames) != 3:
+            print(f"Warning: Only {len(mip_filenames)} replicates were successfully processed")
+            if len(mip_filenames) < 2:
+                print("At least 2 replicates are required for comparison. Exiting.")
+                return 1
+
+        # Now load the MIP files and compare them
+        print("\nLoading MIP files for analysis...")
+        mip_data = []
+        for filename in mip_filenames:
+            try:
+                with mrcfile.open(filename) as mrc:
+                    mip_data.append(mrc.data)
+                    print(f"Loaded {filename}, shape: {mrc.data.shape}, dtype: {mrc.data.dtype}")
+            except Exception as e:
+                print(f"Error loading {filename}: {str(e)}")
+
+        # Check if we have at least 2 MIP files loaded
+        if len(mip_data) < 2:
+            print("Failed to load at least 2 MIP files. Cannot perform comparison.")
+            return 1
+
+        # Verify all threshold values are the same
+        if threshold_values:
+            if not all(abs(v - threshold_values[0]) < 1e-10 for v in threshold_values):
+                print("Warning: Threshold values differ between replicates:")
+                for i, val in enumerate(threshold_values):
+                    print(f"  Replicate {i+1}: {val:.6e}")
+
+            # Use the first threshold value for calculations
+            threshold_value = threshold_values[0]
+        else:
+            print("Warning: No threshold values were extracted.")
+            threshold_value = None
+
+        # Compute similarity metrics between replicates
+        print("\nReproducibility Analysis:")
+        print("========================")
+
+        # Pairwise comparisons of available replicates
+        pairs = [(i, j) for i in range(len(mip_data)) for j in range(i+1, len(mip_data))]
+
+        all_correlations = []
+        all_mean_abs_diffs = []
+        all_psnrs = []
+
+        for i, j in pairs:
```

Comment:
For this section, lets get rid of the psnr calculations. When printing, let's skip the scientific notation and just use the parts per million. When printing the ppm, lets also say what it is relative to, i.e. the threshold value